### PR TITLE
feat: make the CoreAPI expose a streaming pin interface

### PR DESCRIPTION
**Pull request by MichaelMure**
Replace https://github.com/ipfs/interface-go-ipfs-core/pull/38

This PR expose a streamed `pin/ls` API.

### CODE REVIEW

File: tests/block.go [hunk 1]
```diff
@@ -225,9 +225,9 @@ func (tp *TestSuite) TestBlockPin(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	pins, err := api.Pin().Ls(ctx)
+	pins, err := accPins(api.Pin().Ls(ctx))
 	if err != nil {
-		return
+		t.Skip(err)
```
Comment (MichaelMure) [inline tests/block.go]: This is problematic. Before this PR, the test was just terminating and didn't actually test anything.

With this workaround, running the test suite in go-ipfs produce `block.go:230: failed to decode Protocol Buffers: incorrectly formatted merkledag node: unmarshal failed. proto: PBNode: wiretype end group for non-group`.

I don't know what the root cause is.

**Comment by MichaelMure** [comment]
cc @aschmahmann @Stebalien

File: tests/block.go (hunk 1 above)
Comment (Stebalien) [reply tests/block.go]: We're putting a bad block...

**Comment by Stebalien** [comment]
@MichaelMure that should fix your issue. Could you re-test?

**Comment by Stebalien** [comment]
Originally this was blocked on releasing go-ipfs. However, I'm fine adding features to go-ipfs-http-client master and holding off on a release there until there's a go-ipfs release.

**Comment by hsanjuan** [comment]
@Stebalien can this be merged?

**Comment by Stebalien** [comment]
Thanks for bumping this.

**Pull request merged**